test/Parse/recovery.swift

```diff
@@ -44,15 +44,15 @@ func test(a: BadAttributes) -> () { // expected-note * {{did you mean 'test'?}}
 //===--- Recovery for braced blocks.
 
 func braceStmt1() {
-  { braceStmt1(); } // expected-error {{braced block of statements is an unused closure}} expected-error {{expression resolves to an unused function}}
+  { braceStmt1(); } // expected-error {{closure expression is unused}} expected-note {{use 'do' keyword to designate a separate block of statements}} {{3-3=do }}
 }
 
 func braceStmt2() {
-  { () in braceStmt2(); } // expected-error {{expression resolves to an unused function}}
+  { () in braceStmt2(); } // expected-error {{closure expression is unused}} expected-note {{use 'do' keyword to designate a separate block of statements}} {{3-3=do }}
 }
 
 func braceStmt3() {
-  { // expected-error {{braced block of statements is an unused closure}}
```

Comment:
The reason there's no longer a diagnostic in this case is that there was an error typechecking the closure body. I'm not sure how to convince it to produce additional diagnostics in that case...